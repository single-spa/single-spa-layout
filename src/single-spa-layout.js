export { constructRoutes } from "./constructRoutes.js";
export { matchRoute } from "./matchRoute.js";
export { constructApplications } from "./constructApplications.js";
export { constructLayoutEngine } from "./constructLayoutEngine";
