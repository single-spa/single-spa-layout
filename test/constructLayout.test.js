import { constructLayoutEngine } from "../src/single-spa-layout.js";
import { screen } from "@testing-library/dom";

describe(`end-to-end`, () => {
  /** @type {import('../src/constructLayoutEngine').LayoutEngine} */
  let layoutEngine;

  afterEach(() => {
    document.body.innerHTML = "";
    if (layoutEngine) {
      layoutEngine.deactivate();
    }
  });

  it(`can successfully register applications and respond to routing events`, () => {
    /** @type {import('../src/constructRoutes').ResolvedRoutesConfig} */
    const routes = {
      containerEl: "body",
      base: "/",
      mode: "history",
      routes: [
        { type: "application", name: "@org-name/header" },
        {
          type: "route",
          path: "app1",
          routes: [{ type: "application", name: "@org-name/app1" }],
        },
        { type: "application", name: "@org-name/footer" },
      ],
    };

    layoutEngine = constructLayoutEngine({
      routes,
    });

    layoutEngine.activate();

    expect(document.body).toMatchSnapshot();

    history.pushState(history.state, document.title, "/app1");
    window.dispatchEvent(
      new CustomEvent("single-spa:before-mount-routing-event")
    );

    expect(document.body).toMatchSnapshot();
  });
});
