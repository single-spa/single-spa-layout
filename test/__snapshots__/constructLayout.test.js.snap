// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end can successfully register applications and respond to routing events 1`] = `
<body>
  <div
    id="single-spa-application:@org-name/header"
  />
  <div
    id="single-spa-application:@org-name/footer"
  />
</body>
`;

exports[`end-to-end can successfully register applications and respond to routing events 2`] = `
<body>
  <div
    id="single-spa-application:@org-name/header"
  />
  <div
    id="single-spa-application:@org-name/app1"
  />
  <div
    id="single-spa-application:@org-name/footer"
  />
</body>
`;
